packages/block-library/src/image/style.scss

```
@@ -9,7 +9,10 @@
 		max-width: 100%;
 		vertical-align: bottom;
 		box-sizing: border-box;
-		width: fit-content;
+
+		&:not([src$=".svg"]) {
```

Could we be certain that svg files always have `.svg` at the end of their sources in WP?

---

@kevin940726 I believe in the context of the image block, yes. I was researching, and the other option is the `.svgz` extension, which is for compressed SVGs. I believe we should add it, to make it more robust.

```suggestion
		&:not([src$=".svg"]):not([src$=".svgz"]) {
```

---

I'm mostly concerned that SVG files don't have to end with the `.svg` postfix in the URLs. Browsers look for the content types, not the file extensions. I'm not sure if this applies to this case though!

---

I guess it's not certain but highly likely. So if this fixes for 80% of sites then it's an improvement.

---

> I'm mostly concerned that SVG files don't have to end with the .svg postfix in the URLs. Browsers look for the content types, not the file extensions. I'm not sure if this applies to this case though!

This is true, but I believe it doesn't apply to this case, where the SVG is uploaded via UI and inserted via image block. If I'm not mistaken, the media library/image block should insert images with their file extensions.

The problem would be if a block themer, includes an image block on some pattern or template, where the image src uses `data:image/...`, and if the image doesn't have dimensions.

---

First of all, I agree! I think this is a good improvement even though I have some minor concerns. I think we can merge this and iterate if necessary.

> If I'm not mistaken, the media library/image block should insert images with their file extensions.

Would someone be able to insert an SVG file via an external URL? I'm not too familiar with the media upload module, but could someone upload an SVG file without the extension? That is, does the code look for the MIME type or the file extension?

---

> Would someone be able to insert an SVG file via an external URL? I'm not too familiar with the media upload module, but could someone upload an SVG file without the extension?

As far as I know, not via UI using the medialibrary.